nntrainer/include/tensor.h

```diff
@@ -41,6 +41,7 @@ extern "C" {
 
 namespace nntrainer {
 
+class DeferredTensor;
 /**
```

Comment:
how about implement tensor directly. Cause if this works well, we are not going to use tensor any way. Isn't it? and also it looks like duplication to have add and add_i.